code/modules/mob/living/simple_animal/hostile/mining_mobs/ice demon.dm

```diff
@@ -46,20 +46,22 @@
 /obj/projectile/temp/basilisk/ice
 	name = "ice blast"
 	damage = 5
+	speed = 4
 	nodamage = FALSE
 	temperature = -75
 
 /mob/living/simple_animal/hostile/asteroid/ice_demon/OpenFire()
+	ranged_cooldown = world.time + ranged_cooldown_time
 	// Sentient ice demons teleporting has been linked to server crashes
 	if(client)
 		return ..()
 	if(teleport_distance <= 0)
 		return ..()
-	var/list/possible_ends = list()
-	for(var/turf/T in view(teleport_distance, target.loc) - view(teleport_distance - 1, target.loc))
-		if(isclosedturf(T))
+	var/list/possible_ends = view(teleport_distance, target.loc) - view(teleport_distance - 1, target.loc)
+	for(var/turf/T in possible_ends)
+		if(isopenturf(T))
 			continue
```

Comment:
```suggestion
	for(var/turf/closed/T in possible_ends)
```

Then, https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names